ocs_sample_library_preview/SharedStreams.py

```diff
@@ -0,0 +1,482 @@
+from typing import Any
+
+from .BaseClient import BaseClient
+from .SDS.SdsResultPage import SdsResultPage
+from .SDS.SdsStream import SdsStream
+from .SDS.SdsType import SdsType
+from .PatchableSecurable import PatchableSecurable
+
+from .Streams import Streams
+
+class SharedStreams(PatchableSecurable, object):
+    """
+    Client for interacting with Streams shared in a Community
+    """
+
+    def __init__(self, client: BaseClient):
+        """
+        :param client: base client that handles auth and base routing
+        """
+        super().__init__(client=client, collection='Streams')
+
+        self.__uri_api = client.uri_API
+        self.__base_client = client
+
+        # Streams client used to handle requests
+        self.__streams = Streams(self.__base_client)
+
+        self.__setPathAndQueryTemplates()
+
+    def getStream(self, tenant_id: str, namespace_id: str, community_id: str, stream_id: str) -> SdsStream:
+        """
+        Retrieves a stream specified by 'stream_id' in a community specified by 'community_id' from the Sds Service
+        :param tenant_id: tenant to work against
+        :param namespace_id: namespace to work against
+        :param community_id: community to work against
+        :param stream_id: id of the stream
+        :return:the Stream as SdsStream
+        """
+        self.__validateParameters(tenant_id, namespace_id, community_id, stream_id)
+        headers = self.__base_client.communityHeaders(community_id=community_id)
+
+        response = self.__base_client.request(
+            'get',
+            self.__stream_path.format(
+                tenant_id=tenant_id,
+                namespace_id=namespace_id,
+                stream_id=self.__base_client.encode(stream_id)),
+            headers=headers)
+        self.__base_client.checkResponse(
+            response, f'Failed to get SdsStream, {stream_id}.')
+
+        result = SdsStream.fromJson(response.json())
+        return result
+
+    def getStreamType(self, tenant_id: str, namespace_id: str, community_id: str, stream_id: str) -> SdsType:
+        """
+        Retrieves a stream type specified by 'stream_id' in a community specified by 'community_id' from the Sds Service
+        :param tenant_id: tenant to work against
+        :param namespace_id: namespace to work against
+        :param community_id: community to work against
+        :param stream_id: id of the stream
+        :return: the stream type as an SdsType
+        """
+        self.__validateParameters(tenant_id, namespace_id, community_id, stream_id)
+        headers = self.__base_client.communityHeaders(community_id=community_id)
+
+        response = self.__base_client.request(
+            'get',
+            self.__stream_path.format(
+                tenant_id=tenant_id,
+                namespace_id=namespace_id,
+                stream_id=self.__base_client.encode(stream_id)) + '/Type',
```

Comment:
Personally, I'd like to make another template called __stream_type_path or similar, and move it to the bottom. Then either in this PR or a follow up PR later, do the same for all the templates in Streams.py

Reply:
If we do the Streams.py ones in a separate PR (or not do it at all for some reason), then I'm fine not doing this one now, and doing all of them together in one. Since this one is being PR'd though I think it makes sense to change it now if it's an agreeable change

Reply:
I think that's a good observation, I will just do this one here now and we can do the Streams in a separate PR.

Reply:
Updated